xds/src/test/java/io/grpc/xds/XdsClientImplTestBase.java

```diff
@@ -3472,6 +3475,48 @@ public void serverSideListenerResponseErrorHandling_badTransportSocketName() {
     verifyStatusWithNodeId(errorCaptor.getValue(), Code.UNAVAILABLE, errorMsg);
   }
 
+  @Test
+  public void sendingToStoppedServer() throws Exception {
+    try {
+      // Establish the adsStream object
+      xdsClient.watchXdsResource(XdsClusterResource.getInstance(), CDS_RESOURCE,
+          cdsResourceWatcher);
+      resourceDiscoveryCalls.take(); // clear this entry
+
+      // Shutdown server and initiate a request
+      xdsServer.shutdownNow();
+      xdsClient.watchXdsResource(XdsListenerResource.getInstance(), LDS_RESOURCE,
+          ldsResourceWatcher);
+      fakeClock.forwardTime(14, TimeUnit.SECONDS);
```

Comment:
What's the significance of 14 seconds? Is that 1 short of [15 seconds](https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#knowing-when-a-requested-resource-does-not-exist) ? Since the server is/was down even after 15 seconds you shouldn't see "resource does not exist" callback, right (line 3506)? In that case wouldn't it be better for this value to be 15 (or higher) to verify that?

Reply:
We are verifying that the timer gets initialized after the connection is ready rather than when watchXdsResource is called.  I wanted the 15 seconds to terminate between the server coming back up and the response being sent, so have the server down for 14 seconds and then move forward 5 more seconds before doing some checks for no timeouts.  If you feel strongly, I could create an extra test case that has the server down for 16 seconds.

Reply:
Thanks for the explanation. Makes sense